tests/core_tests.py

```diff
@@ -206,6 +205,15 @@ def test_cache_key_changes_when_datasource_is_updated(self):
         # the new cache_key should be different due to updated datasource
         self.assertNotEqual(cache_key_original, cache_key_new)
 
+    def test_cache_key_changes_with_url_parameters(self):
+        self.login(username="admin")
+        slc = self.get_slice("Girls", db.session)
+        viz = slc.viz
+        qobj = viz.query_obj()
+        cache_key_original = viz.cache_key(qobj)
+        viz.form_data["url_params"] = {"foo": "bar"}
+        self.assertNotEqual(cache_key_original, viz.cache_key(qobj))
+
```

Comment:
I would recommend adding a test making sure that the cache key does not change when `url_params` changes, unless some part of the `query_obj` is referencing one of the keys present in `url_params` using Jinja. The point being we want to make sure only the url params that are used affect the cache key.

Reply:
Agreed.